### DISCUSSION TITLE
Get current keyboard layout

### DISCUSSION THREAD

**Discussion by dancaer69** [labwc/labwc#951] [Q&A] [answered]
Is there any way to get the current keyborad layout(I've set layouts in environment and toggle them with alt_shift)? I searched about it and find only a way which works only in sway with swahmsg.

**Comment by stefonarch** [comment, marked as answer]
Afaik there is none atm, but if you use only 2 layouts and don't use the lock scroll indicator there is hack I'm using, in environment:

`XKB_DEFAULT_OPTIONS=grp:alt_shift_toggle,grp_led:scroll`

In lxqt-panel/custom-plugin I'm using this bash line:

`if [ "$(cat /sys/class/leds/input3\:\:scrolllock/brightness)" == "1" ]; then echo ' CH';else echo ' IT'; fi` and it displays the layout on the panel. `input3` can be different, this in arch, on debian it's `input5`.

**Reply by dancaer69** [reply]
Thanks. In fact I use scrolllock and I already set the grp_led:scroll option in evnironment. I just needed to change to input4(as you mentioned) and seems that I can get the current layout.

**Comment by Consolatis** [comment]
This is a pretty nice workaround.

A bit of background information why its so difficult for panels to create a layout indicator only using official wayland protocols:
Every wayland application gets told the current layout when it receives keyboard focus. However that would mean that if a panel was to use that information for an indicator it would require the user to give the panel keyboard focus as to have the up-to-date information which is just not practical. Most of the time the panel would show some outdated state.

See https://github.com/johanmalm/keyboard-layout for a variant of getting the keyboard layout that way.

**Reply by stefonarch** [reply]
Uhm `keyboard-layout` disabled my keyboard somehow once  launched.

So one way could be having a command to switch layout and a counter?

Both wayfire and hyprland have a [plugin](https://github.com/AlexJakeGreen/wayfire-kbdd-plugin) or [script](https://github.com/coffebar/hyprland-per-window-layout/) which enables per-window-layout, which is one thing I miss  in labwc.

**Comment by stefonarch** [comment]
Tried out https://github.com/drougas/xkb-monitor, it works fine


https://github.com/user-attachments/assets/62d02ca0-2f1e-4dd4-96aa-db18f9533be4

In lxqt-panel, customcommand:
* monospace font
* continuos output
* no repeat
* small size (~20px)

layout:      `xkb-monitor -n`
capslock: `xkb-monitor |jq --unbuffered -R -r 'split(",") | if .[4] == "1" then "C " else "·"  end'`
numlock `xkb-monitor |jq --unbuffered -R -r 'split(",") | if .[5] == "1" then "N " else "·"  end'`


<img width="640" height="826" alt="CUSTOMCOMMAND" src="https://github.com/user-attachments/assets/c3809b01-6e9c-4d2d-8c98-f51a4808fbee" />

**Comment by johanmalm** [comment]
> Tried out https://github.com/drougas/xkb-monitor, it works fine

Great find :smile:
